csp_billing_adapter_local/plugin.py

```diff
@@ -33,6 +33,27 @@
 ADAPTER_DATA_DIR = '/var/lib/csp-billing-adapter'
 CACHE_FILE = 'cache.json'
 CSP_CONFIG_FILE = 'csp-config.json'
+CSP_LOG_FILEPATH = '/var/log/csp_billing_adapter.log'
+
+
+def _set_logger_file():
+    """Set logger file with level and format."""
+    try:
+        local_config_path = Path(ADAPTER_DATA_DIR).joinpath(CSP_CONFIG_FILE)
+        with open(local_config_path, 'r', encoding='utf-8') as f:
+            csp_config = json.load(f)
+        csp_config_logging = csp_config.get('logging', {})
+        logging.basicConfig(
+            filename=CSP_LOG_FILEPATH,
+            level=csp_config_logging.get('level', 'INFO'),
+            format="%(asctime)s %(levelname)s: %(message)s"
+        )
+        log.debug('Logger file set')
+    except (FileNotFoundError, JSONDecodeError) as err:
```

Comment:
So I think the file not found error here is related to the open of the local_config_path, rather than the log file itself... Similarly the JSONDecodeError...
It might be better to move this exception handling to immediately after the with open()... block that it technically applies to, and then include an else block that sets a meaningful fall back for the case where the config wasn't found, and then setup logging with that?